src/target/riscv_debug.c

```diff
@@ -999,7 +1047,34 @@ static size_t riscv_build_target_description(
 	offset += snprintf(buffer + offset, print_size, "<reg name=\"pc\" bitsize=\"%u\"%s/>", address_width,
 		gdb_reg_type_strings[GDB_TYPE_CODE_PTR]);
 
-	/* XXX: TODO - implement generation of the FPU feature and registers */
+	
+
+	/* Basic single precision support */
+	if (extensions & RV_ISA_EXT_SINGLE_FLOAT) {
```

Comment:
Given this needs to essentially be done again for doubles, please move this into its own function above this one. That way it can be re-used more easily to build out the `RV_ISA_EXT_DOUBLE_FLOAT` handling.